core/arch/arm/sm/sm.c

```diff
@@ -93,6 +93,10 @@ uint32_t sm_from_nsec(struct sm_ctx *ctx)
 	sm_restore_unbanked_regs(&ctx->sec.ub_regs);
 
 	memcpy(&ctx->sec.r0, args, sizeof(*args));
+#ifdef CFG_CORE_WORKAROUND_ARM_NMFI
+	/* Make sure FIQ is masked when jumping to SMC entry. */
+	ctx->sec.mon_spsr |= CPSR_F;
```

Comment:
Couldn't it be done once per CPU in `init_sec_mon()` where we currently have `sec_ctx->mon_spsr = CPSR_MODE_SVC | CPSR_I;`?

Reply:
Unfortunately no. In first times we enter in here (and without the feature) the FIQ is masked.

But due to FIQ unmasking and then trying to mask it again (which fails) then it stores back the MON SPSR without the FIQ mask.